File: php/features.php
<?php
declare(strict_types=1);

// CrowdSourcedLyrics SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CrowdSourcedLyricsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CrowdSourcedLyricsBaseFeature();
            case "test":
                return new CrowdSourcedLyricsTestFeature();
            default:
                return new CrowdSourcedLyricsBaseFeature();
        }
    }
}

File: py/features.py
# CrowdSourcedLyrics SDK feature factory

from feature.base_feature import CrowdSourcedLyricsBaseFeature
from feature.test_feature import CrowdSourcedLyricsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CrowdSourcedLyricsBaseFeature(),
        "test": lambda: CrowdSourcedLyricsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
